goshopify.go

```diff
@@ -221,11 +234,22 @@ func (a App) NewClient(shopName, token string, opts ...Option) *Client {
 // token. The shopName parameter is the shop's myshopify domain,
 // e.g. "theshop.myshopify.com", or simply "theshop"
 func NewClient(app App, shopName, token string, opts ...Option) *Client {
-	httpClient := http.DefaultClient
+	baseURL, err := url.Parse(ShopBaseUrl(shopName))
+	if err != nil {
+		panic(err) // something really wrong with shopName
```

Comment:
I'm torn about a panic here but totally agree it's better to check it than just ignore it.

Reply:
you reeeeeeally need to screw up to get this... i dont disagree panic is lame but adding an error return seemed like a pain in the ass, if you want the panic gone i'd suggest we go back to ignoring the error